package.json

```
@@ -53,7 +53,7 @@
     "remark-preset-wooorm": "^6.0.0",
     "tape": "^4.0.0",
     "tinyify": "^2.0.0",
-    "xo": "^0.27.0"
+    "xo": "0.27.0"
```

```suggestion
    "xo": "^0.27.0"
```

Could you revert this change?

---

I pushed updates to the dev dependencies so this should no longer be an issue

---

This change is fine with me. It seems the change you made solved this issue. Still, I see that one of the last test pipelines was not executed successfully. https://travis-ci.org/github/vfile/vfile/builds/671479197?utm_source=github_status&utm_medium=notification

---

CI failures seem to be unrelated: Both GH and Travis fail on installing Node for some reason.

---

About the other question of not defining the "result" key at all. The description of the data key is misleading, in my opinion. Instead of writing data?: unknown and setting the type to "unknown," you could have set the type to "any." Hiding information behind an even more complex data structure is then possible. The only option you have in typescript to not define a property (key in an object) at all is to add "[key: string]: any" to an interface VFile the same as in VFileOptions. But there are problems with that approach, some of with Intelli sense of IDEs. Also, typescript set to a stricter mode wouldn't probably allow this kind of declaration.

---

Yes, the description of allowing anything directly on `file` should indeed not be in `file.data`. I think a new documentation section defining common stuff set on `file`, like how it’s done in `vfile-message` in its [Custom properties](https://github.com/vfile/vfile-message#custom-properties) section, makes sense.
I can take that on and do docs later!

I’m indeed thinking of `[key: string]: any` on `file`. This is already what passing a random field in `VFileOptions` will result in.